javascript/app-discover/src/api/api.ts

```diff
@@ -0,0 +1,74 @@
+/*!
+ * Copyright (c) Microsoft. All rights reserved.
+ * Licensed under the MIT license. See LICENSE file in the project.
+ */
+import { sleep } from '../utils/Sleep.js'
+import type {
+	DiscoverProgressCallback,
+	DiscoverStartResponse,
+	DiscoverStatusResponse,
+} from './types.js'
+import { DiscoverResponseStatus } from './types.js'
+
+const RUN_CAUSAL_DISCOVERY_BASE_URL = process.env.DISCOVER_API_URL?.endsWith(
+	'/',
+)
+	? process.env.DISCOVER_API_URL
+	: `${process.env.DISCOVER_API_URL || '/api/discover'}/`
+
+export async function fetchDiscoverResult(
+	algorithmName: string,
+	body: string,
+	progressCallback?: DiscoverProgressCallback,
+): Promise<{ result: unknown; taskId: string }> {
+	const taskId = await startDiscoverAndGetTaskId(algorithmName, body)
+
+	let status = await fetchStatus(taskId)
+
+	progressCallback?.(0, taskId)
+
+	while (isProcessingStatus(status.status)) {
+		await sleep(1000)
```

Comment:
Let's put a TODO to replace polling with something more elegant

Reply:
We were thinking that I could refactor the UI logic in the next days to behave the same way as we have in the exposure app

Reply:
Oh yes, this is just the first approach to validate the backend. I would prefer updating the code to use web sockets.